query.go

```diff
@@ -484,24 +496,35 @@ func queryChainServiceBatch(
 			p := s.PeerByAddr(peer)
 			if p == nil || !p.Connected() {
 				close(quitChan)
-				close(matchSignals[peer])
 				delete(peerQuits, peer)
-				delete(matchSignals, peer)
+
+				// Delete the current state of the peer, so we
+				// won't process any lingering responses from
+				// it.
+				mtxPeerStates.Lock()
+				delete(peerStates, peer)
```

Comment:
Would it be better to do this within `peerGoroutine` so that all modifications to this map happen there?

Reply:
Done: 3484b86